applicationset/services/pull_request/github.go

```diff
@@ -54,6 +56,9 @@ func (g *GithubService) List(ctx context.Context) ([]*PullRequest, error) {
 			PerPage: 100,
 		},
 	}
+	if g.pullRequestState != "" {
+		opts.State = g.pullRequestState
+	}
```

Comment:
Why not set this above unconditionally? If it can override previous value, we probably should combine the values somewhere on a higher level in a well-defined manner.